### DISCUSSION TITLE
React-select with yupResolver, default values, and getOptionValue

### DISCUSSION THREAD

**Discussion by jacobgranberry** [react-hook-form/react-hook-form#2218] [Q&A]
Hi there, could use some help getting this sorted in my form.  First off, heres a codesandbox: https://codesandbox.io/s/react-hook-form-v6-controller-ev0l3?file=/src/index.js

I have a value thats stored as a string in `globalUser`.  

I have a set of options passed to ReactSelect thats _not_ in the standard object format as recommended `({ value: '', label: '' })`, and is instead `{ id: '', label: ''}` so I am using the react-select property `getOptionValue` to tell react-select that the property id is the actual value.  I created a wrapper component around react-select to achieve this (found that answer in another discussion here).

I also have a `yupResolver` to verify that that value set in react-select is an object so I can grab object.id in the `submitHandler` and store the id as a string.

My problem is that I can't figure out how to set the `defaultValue`.  Could really use some guidance here, been banging my head against it for a bit :)  Thanks!

**Comment by bluebill1049** [comment]
you mean set the default select?

**Reply by jacobgranberry** [reply]
Right - the selection in defaultValues doesnt seem to be populating

**Comment by zouterover** [comment]
Same problem here...

**Comment by bluebill1049** [comment]
oops miss the reply, here is a working example for react-select with defaultValue: https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r

**Reply by zouterover** [reply]
Thanks for quick reply!
